src/components/TokenAllowanceGuard/TokenAllowanceGuard.tsx

```diff
@@ -39,47 +39,42 @@ const useStyles = makeStyles<Theme>(theme => ({
 
 export const TokenAllowanceGuard: React.FC<{
   message: ReactNode;
+  isVertical?: boolean;
   tokenAddressMap: AddressMap;
   spenderAddressMap: AddressMap;
-}> = props => {
-  const classes = useStyles();
-  const approveMutation = useApproveToken(props.tokenAddressMap, props.spenderAddressMap);
-  const { data: allowance } = useContractAllowance(props.tokenAddressMap, props.spenderAddressMap);
+}> = ({ message, isVertical = false, tokenAddressMap, spenderAddressMap, children }) => {
+  const approveMutation = useApproveToken(tokenAddressMap, spenderAddressMap);
+  const { data: allowance } = useContractAllowance(tokenAddressMap, spenderAddressMap);
 
   if (!allowance)
     return (
-      <Grid container className={classes.inputRow}>
+      <Box display="flex" alignItems="center" justifyContent="center" height={isVertical ? "84px" : "40px"}>
         <Skeleton width="150px" />
-      </Grid>
+      </Box>
     );
 
   if (allowance.eq(0))
     return (
-      <Grid container className={classes.inputRow}>
-        <Grid item xs={12} sm={8} className={classes.gridItem}>
-          <Box sx={{ display: "flex", alignItems: "center", justifyContent: "center" }}>
-            <Typography variant="body1" className="stake-note" color="textSecondary">
-              {props.message}
+      <Grid container>
+        <Grid item xs={12} sm={isVertical ? 12 : 8}>
+          <Box display="flex" textAlign="center" alignItems="center" justifyContent="center">
+            <Typography variant="body1" color="textSecondary">
+              <em>{message}</em>
             </Typography>
           </Box>
         </Grid>
 
-        <Grid item xs={12} sm={4} className={classes.gridItem}>
-          <Box sx={{ marginTop: { xs: 1, sm: 0 } }}>
-            <PrimaryButton
-              fullWidth
-              className={classes.button}
-              onClick={approveMutation.mutate}
-              disabled={approveMutation.isLoading}
-            >
+        <Grid item xs={12} sm={isVertical ? 12 : 4}>
+          <Box display="flex" alignItems="center" justifyContent="center" mt={[2, isVertical ? 2 : 0]}>
+            <PrimaryButton fullWidth className="" onClick={approveMutation.mutate} disabled={approveMutation.isLoading}>
               {approveMutation.isLoading ? "Approving..." : "Approve"}
             </PrimaryButton>
           </Box>
         </Grid>
       </Grid>
     );
 
-  return <>{props.children}</>;
+  return <>{children}</>;
 };
```

Comment:
With the Give react-query changes just merged. Can we refactor this a bit more to remove the 'GiveTokenAllowanceGuard' ? Looks like they're mostly the same minus some styling.

Reply:
Since they do pretty much differ only in their styling, it makes abstracting and merging these non-trivial and will just make the code more complex (and likely add more lines that it'll save). So it's probably simpler if we just keep it as is for now

Reply:
Also I'll be putting forward some ideas for improving the UI/UX of contract approvals soon, which would involve scrapping these components entirely

Reply:
Sweet!